lib/instrumentation/modules/memcached.js

```diff
@@ -14,16 +15,31 @@ module.exports = function (memcached, agent, { version, enabled }) {
   }
   agent.logger.debug('shimming memcached.prototype.command')
   shimmer.wrap(memcached.prototype, 'command', wrapCommand)
+  shimmer.wrap(memcached.prototype, 'connect', wrapConnect)
   return memcached
 
+  function wrapConnect (original) {
+    return function wrappedConnect () {
+      const currentSpan = agent._instrumentation.currentSpan
+      const server = arguments[0]
+      agent.logger.debug('intercepted call to memcached.prototype.connect %o', { server })
+
+      if (currentSpan) {
+        const [host, port = 11211] = server.split(':')
+        currentSpan.setDestinationContext(getDBDestination(currentSpan, host, port))
+      }
+      return original.apply(this, arguments)
+    }
+  }
+
   // Wrap the generic command that is used to build touch, get, gets etc
   function wrapCommand (original) {
     return function wrappedCommand () {
       if (typeof arguments[0] === 'function') {
         var query = arguments[0]()
         // If the callback is not a function the user doesn't care about result
         if (query && typeof query.callback === 'function') {
-          var span = agent.startSpan(`memcached.${query.type}`, `db.memcached.${query.type}`)
+          var span = agent.startSpan(`memcached.${query.type}`, 'db', 'memcached', `${query.type}`)
```

Comment:
No need for the double wrapped string right? Or is this in case it's not a string?

```suggestion
          var span = agent.startSpan(`memcached.${query.type}`, 'db', 'memcached', query.type)
```

Reply:
Yes indeed 👍